lib/housekeeping.js

```diff
@@ -38,17 +38,109 @@ class HouseKeeper {
     this.tagger = tagger;
   }
 
-  async _handleVolumeData(volume) {}
+  _calculateVolumeTotals(volumes) {}
+
+  _submitVolumeTotals(totalsByCategory) {
+    // Assumption:
+    // totalsByCategory is organized as follows:
+    // { 
+    //   ${region}: [ 
+    //     {
+    //       ${vol-type}: {
+    //         active: {
+    //           gb: value,
+    //           count: value
+    //         },
+    //         unused: {
+    //           gb: value,
+    //           count: value
+    //         }
+    //       }
+    //     },
+    //     ...
+    //   ]
+    // }
+
+    //assert(typeof totalsByCategory === 'object');
+
+    //assert(Object.keys(totalsByCategory).length === 1);
+    let region =  Object.keys(totalsByCategory)[0];
+    
+    let volumeTypes = totalsByCategory[region];
+    //assert(typeof volumeTypes === 'object');
+    //assert(typeof volumeTypes.hasOwnProperty('length'));
+
+    volumeTypes.forEach(function(data) {
+      //assert(typeof data === 'object');
+      
+      //assert(Object.keys(data).length === 1);
+      let type = Object.keys(data)[0];
+
+      //assert(data.hasOwnProperty('active'));
+      //assert(data.hasOwnProperty('unused'));
+      let active = data.active;
+      let unused = data.unused;
+
+      //assert(typeof active === 'object');
+      //assert(active.hasOwnProperty('gb'));
+      //assert(active.hasOwnProperty('count'));
+
+      this.monitor.count('ebs-volumes:${region}:gb-usage', active.gb);
+      this.monitor.count('ebs-volumes:${region}:count', active.count);
+
+      //assert(typeof unused === 'object');
+      //assert(unused.hasOwnProperty('gb'));
+      //assert(unused.hasOwnProperty('count'));
+
+      this.monitor.count('ebs-volumes:${region}:unused-gb-usage', unused.gb);
+      this.monitor.count('ebs-volumes:${region}:unused-count', unused.count);
+
+      let totalGbUsage = active.gb + unused.gb;
+      let totalCount = active.count + unused.count;
+      this.monitor.count('ebs-volumes:global:gb-usage', totalGbUsage);
```

Comment:
And similarly, should this totalGbUsage report happen outside the forEach loop?

Reply:
You're very right, that makes a lot more sense.

I'll still need to rely on the following note from the statsum repo, which leads to the question I mentioned in the PR comments:
> As a rule of thumb, it is reasonable to accumulate metrics for 30 to 90 seconds before flushing. As statsum will only reports every 5min.